modules/web-console/src/main/java/org/apache/ignite/console/routes/AgentDownloadRouter.java

```diff
@@ -0,0 +1,149 @@
+
+    /** */
+    private final String file;
+
+    /**
+     * @param ignite Ignite.
+     */
+    public AgentDownloadRouter(Ignite ignite, String folder, String file) {
+        super(ignite);
+
+        this.file = file;
+        this.pathToWebAgentZip = Paths.get(folder, file + ".zip").toFile();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void initializeCaches() {
+        // No-op so far.
+    }
+
+    /** {@inheritDoc} */
+    @Override public void install(Router router) {
+        router.get("/api/v1/downloads/agent").handler(this::load);
+    }
+
+    /**
+     * @param ctx Context.
+     */
+    private void load(RoutingContext ctx) {
+        User user = checkUser(ctx);
+
+        if (user != null) {
+            try {
+                ZipFile zip = new ZipFile(pathToWebAgentZip);
+
+                ByteArrayOutputStream baos = new ByteArrayOutputStream(BUFFER_SZ);
+
+                ZipOutputStream zos = new ZipOutputStream(baos);
+
+                Enumeration<? extends ZipEntry> entries = zip.entries();
+
+                byte[] buf = new byte[BUFFER_SZ];
+
+                while (entries.hasMoreElements()) {
+                    ZipEntry e = entries.nextElement();
+
+                    zos.putNextEntry(e);
+
+                    if (!e.isDirectory()) {
+                        int bytesRead;
+
+                        InputStream is = zip.getInputStream(e);
+
+                        while ((bytesRead = is.read(buf)) != -1)
+                            zos.write(buf, 0, bytesRead);
+                    }
+
+                    zos.closeEntry();
+                }
+
+                // Append default.properties to ZIP.
+                ZipEntry e = new ZipEntry(file + "/default.properties");
+                zos.putNextEntry(e);
+
+                String content = String.join("\n",
+                    "tokens=" + user.principal().getString("token", "MY_TOKEN"),
+                    "server-uri=" + ctx.request().host(),
+                    "#Uncomment following options if needed:",
+                    "#node-uri=http://localhost:8080",
+                    "#node-login=ignite",
+                    "#node-password=ignite",
+                    "#driver-folder=./jdbc-drivers",
+                    "#Uncomment and configure following SSL options if needed:",
+                    "#node-key-store=client.jks",
+                    "#node-key-store-password=MY_PASSWORD",
+                    "#node-trust-store=ca.jks",
+                    "#node-trust-store-password=MY_PASSWORD",
+                    "#server-key-store=client.jks",
+                    "#server-key-store-password=MY_PASSWORD",
+                    "#server-trust-store=ca.jks",
+                    "#server-trust-store-password=MY_PASSWORD",
+                    "#cipher-suites=CIPHER1,CIPHER2,CIPHER3"
+                );
+
+                zos.write(content.getBytes());
+                zos.closeEntry();
+                zos.close();
```

Comment:
Please check that executable bits is stay on .sh file after download.

Reply:
Fixed by using apache-commons-compress